data_model/src/events/trigger_executed.rs

```diff
@@ -0,0 +1,255 @@
+//! Event that notifies some trigger was somehow executed.
+
+#[cfg(not(feature = "std"))]
+use alloc::{format, string::String, vec::Vec};
+
+use iroha_ffi::FfiType;
+use iroha_macro::FromVariant;
+use iroha_schema::prelude::IntoSchema;
+use parity_scale_codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+
+use super::Filter;
+use crate::prelude::{Metadata, TriggerId};
+
+/// [`Event`] filter.
+#[derive(
+    Default,
+    Debug,
+    PartialOrd,
+    Ord,
+    PartialEq,
+    Eq,
+    Clone,
+    Decode,
+    Encode,
+    IntoSchema,
+    Hash,
+    Serialize,
+    Deserialize,
+    FfiType,
+)]
+pub struct EventFilter {
+    /// If `Some`, filters by the [`TriggerId`]. Otherwise accepts any [`TriggerId`].
+    pub id: Option<TriggerId>,
+    /// If `Some`, filters by the [`StatusKind`]. Otherwise accepts any [`StatusKind`].
+    pub status_kind: Option<StatusKind>,
```

Comment:
We could go without `StatusKind` if event `TriggerExecuted` is only be emitted if trigger was successfully executed. We don't seem to emit failed events(only successful) for any other type of fallible event. Consider following this convention

Reply:
The `Failed` variant would correspond to `Rejected` of pipeline events. Otherwise what will be a notification of trigger execution failures?